smart-contracts/contracts/BridgeBank/BridgeBank.sol

```diff
@@ -484,43 +623,47 @@ contract BridgeBank is BankStorage,
         );
     }
 
+    /**
+     * @dev Burns a token
+     * @param recipient Bytes representation of destination address
+     * @param tokenAddress Token address
+     * @param tokenAmount How much should be burned
+     * @param _lockBurnNonce A global nonce for locking an burning tokens
+     */
     function _burnTokens(
         bytes calldata recipient,
         address tokenAddress,
         uint256 tokenAmount,
         uint256 _lockBurnNonce
     ) private onlyCosmosTokenWhiteList(tokenAddress) validSifAddress(recipient) {
         BridgeToken tokenToTransfer = BridgeToken(tokenAddress);
+        
         // burn tokens
-        tokenToTransfer.burnFrom(
-            msg.sender,
-            tokenAmount
-        );
+        tokenToTransfer.burnFrom(msg.sender, tokenAmount);
+
+        // revert if the token doesn't have a denom
+        string memory denom = getDenom(tokenAddress);
+        require(keccak256(abi.encodePacked(denom)) != keccak256(abi.encodePacked("")), "INV_DENOM");
 
         // decimals defaults to 18 if call to decimals fails
         uint8 decimals = getDecimals(tokenAddress);
 
-        // Get name and symbol
-        string memory name = getName(tokenAddress);
-        string memory symbol = getSymbol(tokenAddress);
```

Comment:
The alternative to the above solution is to get the token symbol and do a string compare using keccak256, and if the strings equal each other, then don't require the denom to be not empty.